### DISCUSSION TITLE
Is asMap().values().stream()... counted as read operation ?

### DISCUSSION THREAD

**Discussion by misanek** [ben-manes/caffeine#657] [Q&A] [answered]
Hello,
 I'm using cache which is not so often written to, but it is read  approximately every minute via  `cache.asMap().values().stream()        .anyMatch(...);` , I also have removalListener to perform additional logic.
 Documentation on [CleanUp ](https://github.com/ben-manes/caffeine/wiki/Cleanup) states: 
` Instead, it performs small amounts of maintenance work after write operations or occasionally after read operations if writes are rare. `
So my question is whether `cache.asMap().values().stream().anyMatch(...);` is counted as read and thus I don't need to worry about clean up or should I rather use `Scheduler` as explained in cleanUp docs.

**Comment by ben-manes** [comment, marked as answer]
Iterators don’t count as reads to avoid degrading the eviction policy (access time, frequency, etc). If it observes an expired entry it will skip it, but as of #487 it will also trigger a maintenance cycle to be a bit more proactive. I think you’ll be fine if you are on a semi recent version (v2.9+).

**Reply by misanek** [reply]
Thanks :)
